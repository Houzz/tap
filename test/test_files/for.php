<?php

for ($x = 0 , $y = 0 ; $x < 3 ; ++$x, ++$y) {
    $z = 10;
}

__tap_check($y, "int");
__tap_check($z, "int|null");

for ( ; ; ) {}
for ($x = 0 ; ; ++$x) {}
for ($x = 0 ; $x ; ++$x) {}  // wrong, $x need to be bool

