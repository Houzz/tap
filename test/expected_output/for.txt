8 E_NAME_UNDEFINED
12 E_EXPECTED_TYPE
